### ISSUE TITLE
Add setting to switch from reporting on a file to reporting on the workspace 

### ISSUE THREAD

**Issue by karthiknadig** [microsoft/vscode-mypy#81]

**Comment by MartinBernstorff** [comment]
Would this be equivalent to linting across files?

E.g. if I write this implementation in `abc.py`:
```python
from abc import ABC, ABCMeta, abstractmethod
from typing import Protocol, Tuple


class TestABC(ABC):
    @abstractmethod
    def test(self, testparam2: int) -> float:
        ...


class ABCImplementation(TestABC):
    def test(self, testparam2: str) -> str:
        return testparam2
```

I get an error `Return type "str" of "test" incompatible with return type "float" in supertype "TestABC"Mypy[override](https://mypy.readthedocs.io/en/latest/_refs.html#code-override)`.

But if I move TestABC to another file and import it, I don't get the error. E.g:

```python
from TestABC import TestABC


class ABCImplementation(TestABC):
    def test(self, testparam2: str) -> str:
        return testparam2


def test1(param1: TestABC) -> None:
    param1.test(testparam2=1)
```

**Comment by karthiknadig** [comment]
> Would this be equivalent to linting across files?

That is correct.

**Comment by karrtikr** [comment]
Adding milestone to make sure this is verified.
